src/client/common/process/environmentActivationService.ts

```diff
@@ -143,9 +143,14 @@ export class EnvironmentActivationService implements IEnvironmentActivationServi
         const title = DataScience.activatingPythonEnvironment().format(
             interpreter.displayName || getDisplayPath(interpreter.path)
         );
-        return KernelProgressReporter.wrapAndReportProgress(resource, title, () =>
-            this.getActivatedEnvironmentVariablesImpl(resource, interpreter)
-        );
+        const cancellationTokenSource = new CancellationTokenSource();
+        try {
+            return await KernelProgressReporter.wrapAndReportProgress(resource, title, cancellationTokenSource, (_t) =>
```

Comment:
I agree with @amunger If we didn't have to pass the token this code would be simpler, now we need to pass a token everytime even when not required.